build-test.cmd

```diff
@@ -6,8 +6,17 @@ set __BuildArch=x64
 set __VCBuildArch=x86_amd64
 set __BuildType=Debug
 set __BuildOS=Windows_NT
-set __VSVersion=vs2015
-set __VSToolsRoot=%VS140COMNTOOLS%
+
+:: Default to highest Visual Studio version available
+if defined VS150COMNTOOLS (
```

Comment:
CoreCLR build should be done from a barebones command prompt - this is why VCVarsall.bat is invoked so that it sets the right environment. We cannot (and should not) assume a specific command prompt to be set.

As a result, from a clean command prompt, there is no VS 2017 variables defined and thus, no opted in. The only opt-in works when we explicitly specify one.